src/stepFunctions/commands/publishStateMachine.ts

```diff
@@ -20,20 +21,30 @@ import {
     PublishStateMachineWizardUpdateResponse,
 } from '../wizards/publishStateMachineWizard'
 
+import { VALID_SFN_PUBLISH_FORMATS, YAML_FORMATS } from '../constants/aslFormats'
+
 const DEFAULT_REGION: string = 'us-east-1'
 
 export async function publishStateMachine(awsContext: AwsContext, outputChannel: vscode.OutputChannel) {
     const logger: Logger = getLogger()
 
     const textDocument = vscode.window.activeTextEditor?.document
+
     if (!textDocument) {
         logger.error('Could not get active text editor for state machine definition')
         throw new Error('Could not get active text editor for state machine definition')
     }
 
-    if (textDocument.languageId === 'asl-yaml') {
-        logger.error('Cannot publish state machine from Amazon States Language YAML file')
-        throw new Error('Cannot publish state machine from Amazon States Language YAML file')
+    let text = textDocument.getText()
+
+    if (!VALID_SFN_PUBLISH_FORMATS.includes(textDocument.languageId)) {
+        const errorMessage = `Cannot publish state machine from "${textDocument.languageId}" file`
+        logger.error(errorMessage)
+        throw new Error(errorMessage)
+    }
+
+    if (YAML_FORMATS.includes(textDocument.languageId)) {
+        text = JSON.stringify(safeLoad(text), undefined, '  ')
```

Comment:
Might be able to condense this into the assignment (that way you can also use a `const`):
```
const text = YAML_FORMATS.includes(textDocument.languageId) ? JSON.stringify(safeLoad(textDocument.getText()), undefined, ' ') : textDocument.getText()
```

Reply:
Should any of this be try/caught?

Reply:
I realized that in case of invalid YAML user would get an error message that doesn't say much about nature of the problem. I added functionality to catch error and display message to the user that YAML is invalid.